src/UKHO.ERPFacade.API/Filters/CorrelationIdMiddleware.cs

```diff
@@ -1,28 +1,53 @@
-﻿using System.Diagnostics.CodeAnalysis;
+﻿using Newtonsoft.Json.Linq;
+using System.Diagnostics.CodeAnalysis;
 
 namespace UKHO.ERPFacade.API.Filters
 {
     [ExcludeFromCodeCoverage]
-    public static class CorrelationIdMiddleware
+    public class CorrelationIdMiddleware
     {
-        public const string XCorrelationIdHeaderKey = "X-Correlation-ID";
+        public const string XCorrelationIdHeaderKey = "_X-Correlation-ID";
+        private readonly RequestDelegate _next;
 
-        public static IApplicationBuilder UseCorrelationIdMiddleware(this IApplicationBuilder builder)
+        public CorrelationIdMiddleware(RequestDelegate next)
         {
-            return builder.Use(async (context, func) =>
+            _next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext httpContext)
+        {
+            string? correlationId = string.Empty;
+
+            if (httpContext.Request.ContentLength > 0)
+            {
+                httpContext.Request.EnableBuffering();
+
+                var bodyAsText = await new StreamReader(httpContext.Request.Body).ReadToEndAsync();
```

Comment:
StreamReader is disposable. This line should be something like:

                using var streamReader = new StreamReader(httpContext.Request.Body, leaveOpen: true);
                var bodyAsText = await streamReader.ReadToEndAsync();

Reply:
The code has been refactored.